### PR TITLE
ci: setup github actions, configure golangci-lint and fix lint errors

### LINKED ISSUE #15: goreleaser & github actions for releases
- More transparent than docker cloud build
- Automatically creates release notes
- alignment with other projects

### PR DESCRIPTION
**Pull request by ghouscht**
fixes #15

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      - experimental
+      - opinionated
+      - performance
+      - style
+    disabled-checks:
+      - whyNoLint
+  gocyclo:
+    min-complexity: 15
+  goimports:
+    local-prefixes: github.com/ewohltman/ephemeral-roles
```
Comment (djboris9) [inline .golangci.yml:29]: Why is this needed?

**Review by djboris9 — CHANGES_REQUESTED** [review]

File: .golangci.yml (hunk 1 above)
Comment (ghouscht) [reply .golangci.yml:29]: good question, copy paste from another project... will take a look at it

**Review by djboris9 — APPROVED** [review]

**Comment by djboris9** [comment]
:beers: :beers:
